src/ImageAxes.jl

```diff
@@ -27,6 +27,10 @@ export # types
     timeaxis,
     timedim
 
+ImageCore.HasDimNames(::Type{A}) where {A<:AxisArray} = HasDimNames{true}()
+ImageCore.namedaxes(a::AxisArray) = NamedTuple{axisnames(a)}(axisvalues(a))
+Base.names(a::AxisArray) = axisnames(a)
```

Comment:
This is kind of iffy but it is necessary to really fulfill the intended meaning of `HasDimNames`.

Reply:
Remind me why the name of this function needs to be `names`? (Link to other package, for example?) The reason I ask:
```julia
julia> t = (x = 1, y = 2)
(x = 1, y = 2)

julia> typeof(t)
NamedTuple{(:x, :y),Tuple{Int64,Int64}}

julia> names(t)
ERROR: MethodError: no method matching names(::NamedTuple{(:x, :y),Tuple{Int64,Int64}})
Closest candidates are:
  names(::Module; all, imported) at reflection.jl:98
Stacktrace:
 [1] top-level scope at REPL[3]:1
```
`struct`, for example, uses `fieldnames` rather than just `names`. `axisnames` seems like a really good name for this.

Reply:
It's what NamedDims uses to access the axis names https://github.com/invenia/NamedDims.jl/blob/master/src/wrapper_array.jl#L77.

But hadn't even considered using a different name. `axisnames` would certainly be more descriptive.

Reply:
Let's see what happens in https://github.com/invenia/NamedDims.jl/issues/22